ee/WFLY-1160-Provide_ability_to_configure_a_global_directory_which_puts_the_contents_in_the_classpath.adoc

```diff
@@ -0,0 +1,86 @@
+= Provide ability to easily apply certain JBoss module libraries to all deployments running in a server
+:author:            Yeray Borges
+:email:             yborgess@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+Wildfly users can share libraries across all deployed applications without having to modify the packaging of their applications creating a custom module and making it global using the EE subsystem:
+
+  [standalone@localhost:9990 /] module add --name=com.test --absolute-resources=/Users/me/libs:/Users/me/libs/my-utils1.jar:/Users/me/libs/my-utils2.jar
+  [standalone@localhost:9990 /] /subsystem=ee:list-add(name=global-modules,value={name=com.test})
+
+This configuration implies the following:
+
+  * Users have to specify all required resource roots using --absolute-resources argument.
+  * It is necessary to modify the generated module.xml, or at least re-create the global module, if we want to add/remove any additional resource-root for this global module.
+  * This configuration allows any customization of the generated module.xml, making it very flexible to cover broader cases, for example, it is possible to declare module dependencies, mark them as exported, declare the main class and so on.
+
+This RFE is opened to address more simple cases, where it is not required any additional configuration beyond on the default module.xml configurations.
+
+The simplest way to accomplished this is to use the EE subsystem to configure a path to the shared folder. An example of this configuration could be:
+
+  [standalone@localhost:9990 /] /subsystem=ee:list-add(name=global-folders,value={name=my-common-libs, path=/Users/me/libs})
+
+The name and path attributes will be mandatory.
+Internally, this operation will create a module with the name `my-common-libs` that will have as resource-roots the folder `/Users/me/libs/` and all the jars files included in this folder. Subdirectories will not be scanned, but if it is required, users can create a new global-folder using the subdirectory path.
+
+Removal of a global-folder could be done using the following operation:
+
+  [standalone@localhost:9990 /] /subsystem=ee:list-remove(name=global-folders,value={name=my-common-libs})
+
+This RFE will not cover the availability of hot JBoss modules redeployments, that means, modifications of existing jar files as well as adding a new jar or removal an existing one will require a complete server restart, however changes in plain resources as property files will be available immediately to the applications.
+
```

Comment:
Can you please describe when while server starting it will be shared library loaded?

Reply:
Added:

> The module created from the shared folder will be loaded as soon as the first application is deployed in the server

Reply:
Look good.
Thank you @yersan